colossus/src/main/scala/colossus/protocols/memcache/Memcache.scala

```diff
@@ -189,28 +204,36 @@ sealed trait MemcacheWriteCommand extends MemcacheCommand {
   def value: ByteString
   def ttl: Int
   def flags: Int
+  def casUniqueMaybe: Option[Int]
 
   def bytes(compressor: Compressor): ByteString = {
 
     val b = new ByteStringBuilder
 
     /*write commands are of the format
     We don't support the [noreply] semantics at this time
-    <COMMAND> <KEY> <FLAGS> <EXPTIME> <BYTECOUNT>\r\n<data>\r\n
+    <COMMAND> <KEY> <FLAGS> <EXPTIME> <BYTECOUNT> [<CAS UNIQUE>]\r\n<data>\r\n
 
     padding accounts for spaces, \r\n's, bytecount and exptime:
-       4 spaces
+       4 or 5 spaces, depending if <CAS UNIQUE> is present
     +  4 (2 \r\n's)
      ----
-       8 */
-
-    val padding = 8
+       8 or 9, depending if <CAS UNIQUE> is present*/
 
     val flagsStr    = ByteString(flags.toString)
     val ttlStr      = ByteString(ttl.toString)
     val dataSizeStr = ByteString(value.size.toString)
+    val casUniqueStr = ByteString(casUniqueMaybe.getOrElse("").toString)
+
+    val padding = 8 + (if (casUniqueStr.nonEmpty) 1 else 0)
```

Comment:
Please create a function that return the padding and perhaps mention where this magic 8 comes from.

Reply:
the padding is explained 7 lines up (https://github.com/tumblr/colossus/pull/673/files#diff-102c569d28658e91e688d75f0650d38fR223)

Reply:
Regardless, you should do extract method. This method is already too long. Then you can extract the comment and put it where this new method is.